extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -949,6 +952,10 @@ export class PublishDatabaseDialog {
 
 			this.profileUsed = true;
 			this.publishProfileUri = filePath;
+
+			this.profileSaved = true;
+
+			void this.project.addPublishProfileToProjFile(filePath.fsPath);		//add profile to the sqlproj in background
```

Comment:
why not use await here? Just in case the publish profile doesn't get added to the sqlproj before the tree is refreshed

Reply:
Added it here, and called refresh after this.